quipucords/fingerprinter/task.py

```diff
@@ -393,9 +398,8 @@ def _process_details_report(self, manager_interrupt, details_report):
 
                         facts = {'namespace': 'qpc', 'facts': nested_facts}
                         insights_host['facts'] = [facts]
-                        if bool(nested_facts):
-                            insights_host['system_profile'] = \
-                                self.format_system_profile(fingerprint_dict)
+                        insights_host['system_profile'] = \
+                            self.format_system_profile(fingerprint_dict)
```

Comment:
What happens if `self.format_system_profile(fingerprint_dict)` returns an empty dictionary? would assigning `system_profile={}` in our host be accepted by the inventory?

Reply:
I added code to filter an empty system_profile